sys-cluster/openmpi/openmpi-3.1.1.recipe

```diff
@@ -0,0 +1,228 @@
+
+libmca_common_smVersion=40.10.0
+libmca_common_smVersionCompat="$libmca_common_smVersion compat >= ${libmca_common_smVersion%%.*}"
+
+libmpi_mpifhVersion=40.10.1
+libmpi_mpifhVersionCompat="$libmpi_mpifhVersion compat >= ${libmpi_mpifhVersion%%.*}"
+
+libmpi_usempi_ignore_tkrVersion=40.10.1
+libmpi_usempi_ignore_tkrVersionCompat="$libmpi_usempi_ignore_tkrVersion compat >= ${libmpi_usempi_ignore_tkrVersion%%.*}"
+
+libmpi_usempif08Version=40.10.1
+libmpi_usempif08VersionCompat="$libmpi_usempif08Version compat >= ${libmpi_usempif08Version%%.*}"
+
+libmpiVersion=40.10.1
+libmpiVersionCompat="$libmpiVersion compat >= ${libmpiVersion%%.*}"
+
+libompitraceVersion=40.10.0
+libompitraceVersionCompat="$libompitraceVersion compat >= ${libompitraceVersion%%.*}"
+
+libopen_palVersion=40.10.1
+libopen_palVersionCompat="$libopen_palVersion compat >= ${libopen_palVersion%%.*}"
+
+libopen_rteVersion=40.10.1
+libopen_rteVersionCompat="$libopen_rteVersion compat >= ${libopen_rteVersion%%.*}"
+
+GLOBAL_WRITABLE_FILES="
+	settings/openmpi-default-hostfile keep-old
+	settings/openmpi-mca-params.conf keep-old
+	settings/openmpi-totalview.tcl keep-old
+	settings/pmix-mca-params.conf keep-old
+	"
+
+PROVIDES="
+	openmpi$secondaryArchSuffix = $portVersion
+	cmd:aggregate_profile.pl
+	cmd:mpicc
+	cmd:mpic++
+	cmd:mpicc
+	cmd:mpicxx
+	cmd:mpiexec
+	cmd:mpif77
+	cmd:mpif90
+	cmd:mpifort
+	cmd:mpirun
+	cmd:ompi_clean
+	cmd:ompi_dvm
+	cmd:ompi_ps
+	cmd:ompi_server
+	cmd:ompi_top
+	cmd:ompi_info
+	cmd:opal_wrapper
+	cmd:orte_clean
+	cmd:orte_dvm
+	cmd:orte_info
+	cmd:orte_ps
+	cmd:orte_server
+	cmd:orte_top
+	cmd:ortecc
+	cmd:orted
+	cmd:orterun
+	cmd:profile2mat.pl
+	cmd:prun
+	lib:libmca_common_monitoring$secondaryArchSuffix = $libmca_common_monitoringVersionCompat
+	lib:libmca_common_ompio$secondaryArchSuffix =$libmca_common_ompioVersionCompat
+	lib:libmca_common_sm$secondaryArchSuffix = $libmca_common_smVersionCompat
+	lib:libmpi$secondaryArchSuffix = $libmpiVersionCompat
+	lib:libmpi_mpifh$secondaryArchSuffix = $libmpi_mpifhVersionCompat
+	lib:libmpi_usempi_ignore_tkr$secondaryArchSuffix = $libmpi_usempi_ignore_tkrVersionCompat
+	lib:libmpi_usempif08$secondaryArchSuffix = $libmpi_usempif08VersionCompat
+	lib:libompitrace$secondaryArchSuffix = $libompitraceVersionCompat
+	lib:libopen_pal$secondaryArchSuffix = $libopen_palVersionCompat
+	lib:libopen_rte$secondaryArchSuffix = $libopen_rteVersionCompat
+	lib:ompi_monitoring_prof$secondaryArchSuffix
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	lib:libevent_2.1$secondaryArchSuffix
+	lib:libevent_pthreads$secondaryArchSuffix
+	lib:libgfortran$secondaryArchSuffix
+	lib:libgomp$secondaryArchSuffix
+	lib:libhwloc$secondaryArchSuffix
+	lib:libz$secondaryArchSuffix
+	"
+
+PROVIDES_devel="
+	openmpi${secondaryArchSuffix}_devel = $portVersion
+	devel:libmca_common_monitoring$secondaryArchSuffix = $libmca_common_monitoringVersionCompat
+	devel:libmca_common_ompio$secondaryArchSuffix =$libmca_common_ompioVersionCompat
```

Comment:
missing white space after the `=`:
```diff
- 	devel:libmca_common_ompio$secondaryArchSuffix =$libmca_common_ompioVersionCompat 
+	devel:libmca_common_ompio$secondaryArchSuffix = $libmca_common_ompioVersionCompat 
```

BTW, you might wish to do something similar to what we have for util_linux:

```diff
+openmpiLibs="
+	libbmca_common_monitoring
+	libmca_common_ompio
+	libmca_common_sm
+	libmpi_mpifh
+	libmpi_usempi_ignore_tkr
+	libmpi_usempif08
+	libmpi
+	libompitrace
+	libopen_pal
+	libopen_rte
+	"
 libmca_common_monitoringVersion=50.0.0
-libmca_common_monitoringVersionCompat="$libmca_common_monitoringVersion compat >= ${libmca_common_monitoringVersion%%.*}"
 libmca_common_ompioVersion=41.9.0
-libmca_common_ompioVersionCompat="$libmca_common_ompioVersion compat >= ${libmca_common_ompioVersion%%.*}"
 libmca_common_smVersion=40.10.0
-libmca_common_smVersionCompat="$libmca_common_smVersion compat >= ${libmca_common_smVersion%%.*}"
 libmpi_mpifhVersion=40.10.1
-libmpi_mpifhVersionCompat="$libmpi_mpifhVersion compat >= ${libmpi_mpifhVersion%%.*}"
 libmpi_usempi_ignore_tkrVersion=40.10.1
-libmpi_usempi_ignore_tkrVersionCompat="$libmpi_usempi_ignore_tkrVersion compat >= ${libmpi_usempi_ignore_tkrVersion%%.*}"
 libmpi_usempif08Version=40.10.1
-libmpi_usempif08VersionCompat="$libmpi_usempif08Version compat >= ${libmpi_usempif08Version%%.*}"
 libmpiVersion=40.10.1
-libmpiVersionCompat="$libmpiVersion compat >= ${libmpiVersion%%.*}"
 libompitraceVersion=40.10.0
-libompitraceVersionCompat="$libompitraceVersion compat >= ${libompitraceVersion%%.*}"
 libopen_palVersion=40.10.1
-libopen_palVersionCompat="$libopen_palVersion compat >= ${libopen_palVersion%%.*}"
 libopen_rteVersion=40.10.1
-libopen_rteVersionCompat="$libopen_rteVersion compat >= ${libopen_rteVersion%%.*}"
+for i in $openmpiLibs; do
+	eval "${i}VersionCompat=\"\$${i}Version compat >= \${${i}Version%%.*}\""
+done
```

BTW, you might even be able to pass `$openmpiLibs` to `prepareInstalledDevelLibs`.

Reply:
Reminder: I did not check this, so there might be errors in what I wrote above.